packages/components/src/Checkbox/Checkbox.tsx

```diff
@@ -52,6 +48,22 @@ interface CheckboxProps {
   onChange?(newValue: boolean): void;
 }
 
+interface CheckboxLabelProps extends BaseCheckboxProps {
+  /**
+   * Label that shows up beside the checkbox.
+   */
+  readonly label?: string;
+}
+
+interface CheckboxChildrenProps extends BaseCheckboxProps {
+  /**
+   * Component children, which shows up as a label
+   */
+  readonly children?: JSX.Element;
```

Comment:
we normally use react element for this

```suggestion
  readonly children?: ReactElement;
```